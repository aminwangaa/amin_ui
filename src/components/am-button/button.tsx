import React, {
    AnchorHTMLAttributes,
    ButtonHTMLAttributes,
    PropsWithChildren,
    useMemo
    }
from "react"
import styled from "styled-components";
import StyledButton from "./styleButton"

export type AppearancesTypes = keyof typeof APPEARANCES;

type appearance =
	| "primary"
	| "default"
	| "success"
	| "warning"
	| "danger"

type AppearancesObj = {
	[key in appearance]: appearance;
};

export const APPEARANCES: AppearancesObj = {
	primary: "primary",
	default: "default",
	success: "success",
	warning: "warning",
	danger: "danger",
};

export type SizesTypes = keyof typeof SIZES;
type sizeType = "small" | "medium" | "large";
type sizeObj = {
	[key in sizeType]: sizeType;
};

export const SIZES: sizeObj = {
	small: "small",
	medium: "medium",
	large: "large",
};

type fontSizeObj = {
    [key in sizeType]: string;
};

export const SIZE: fontSizeObj = {
    small: "12px",
    medium: "14px",
    large: "16px",
};

type btnPadding = {
    [key in sizeType]: string;
}

export const btnPadding: btnPadding = {
	small: "6px 12px",
	medium: "8px 16px",
	large: "10px 20px",
};

const Text = styled.span`
	display: inline-block;
	height: 100%;
	width: 100%;
`;

export interface CustormButtonProps {
	/** 按钮类型 */
	appearance?: AppearancesTypes;
	/** 是否需要border */
	border?: boolean;
	/** 是否圆 */
	circle?: boolean;
	/** 自定义类名 */
	className?: string;
	/** 是否禁用 */
	disabled?: boolean;
	/** 是否是a标签 */
	isLink?: boolean;
    /** 是否打开新页面 */
    newPage?: boolean;
	/** 是否圆角 */
	round?: boolean;
	/** 按钮大小 */
	size?: SizesTypes;
    /** a标签时打开方式 */
    target?: string;
	/** a标签时hover有无下划线 */
	textLine?: boolean;
}

export type ButtonProps = CustormButtonProps &
	AnchorHTMLAttributes<HTMLAnchorElement> &
    ButtonHTMLAttributes<HTMLButtonElement>;

export const Button = (props: PropsWithChildren<ButtonProps>) => {
	const {
		isLink,
		children,
        newPage,
		disabled,
	} = props;
	const buttonInner = (
		<Text>{children}</Text>
	);
	const btnType = useMemo(() => {
		if (isLink) {
			return "a";
		} else {
			return "button"
		}
	}, [isLink]);

	type isDisableLink = {
        href?: string;
        target?: string;
        onClick?: (event:React.MouseEvent) => void
    }
	const isDisableLink:isDisableLink = {}
	if (isLink && disabled) {
        isDisableLink.href = "#"
        isDisableLink.onClick = (e) => {
        	e.preventDefault()
			return false
		}
	}

	if (newPage && isLink) {
        isDisableLink.target = "_blank"
	}

	return (
		<StyledButton
			as={btnType}
			{...props}
			{...isDisableLink}
		>
			{buttonInner}
		</StyledButton>
	);
}

Button.defaultProps = {
	isLink: false,
	appearance: APPEARANCES.primary,
	isDisabled: false,
	containsIcon: false,
	size: SIZES.medium,
	border: true,
    circle: false,
    newPage: false,
    round: false,
    disabled: false,
    href: "",
    target: "",
};

export default Button;
