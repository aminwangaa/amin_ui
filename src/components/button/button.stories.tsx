import React from "react";
import { action } from "@storybook/addon-actions";
import  { Button }  from "./index";

// CSF 用法
export default {
	title: "Button",  // 主题 （小标题）
	component: Button,
};

export const Text = () => (  // 导出的 就是 主题下面 的案例
	<Button onClick={action("clicked")}>Hello Button11</Button>
);

export const Emoji = () => (
	<Button onClick={action("clicked")}>hello storybook</Button>
);
