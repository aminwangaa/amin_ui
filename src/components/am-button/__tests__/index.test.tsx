import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Button, SIZES, SIZE, APPEARANCES } from "../button"
import {color, paddingSize} from "../../shared/styles"

describe("test Button component", () => {
    it("size button", () => {
        const sizes = Reflect.ownKeys(SIZES)
        sizes.forEach(item => {
            let wrapper = render(<Button size={item} appearance={"primary"}>{item}</Button>);
            expect(wrapper).toMatchSnapshot();
            let div = wrapper.queryByTestId("button-test");
            expect(div).toHaveStyle(`color: ${color.lightest}`);
            expect(div).toHaveStyle(`background: ${color.primary}`);
            expect(div).toHaveStyle(`border: 1px solid ${color.primary}`);
            expect(div).toHaveStyle(`font-size: ${SIZE[item]}`);
            expect(div).toHaveStyle(`padding: ${paddingSize[item]}`);
            cleanup()
        })
    });

    it("appearance button", () => {
        const appearances = Reflect.ownKeys(APPEARANCES)
        appearances.forEach(item => {
            let wrapper = render(<Button appearance={item}>{item}</Button>);
            expect(wrapper).toMatchSnapshot();
            let div = wrapper.queryByTestId("button-test");
            if (item === "default") {
                expect(div).toHaveStyle(`background:  ${color.lightest}`);
                expect(div).toHaveStyle(`color: ${color.darkest}`);
                expect(div).toHaveStyle(`border: 1px solid ${color.default}`);
            } else {
                expect(div).toHaveStyle(`color: ${color.lightest}`);
                expect(div).toHaveStyle(`background: ${color[item]}`);
                expect(div).toHaveStyle(`border: 1px solid ${color[item]}`);
            }
            cleanup()
        })
    });

    it("round button", () => {
        let wrapper = render(<Button round={true} appearance={"primary"}>primary round</Button>);
        expect(wrapper).toMatchSnapshot();
        let div = wrapper.queryByTestId("button-test");

        expect(div).toHaveStyle(`color: ${color.lightest}`);
        expect(div).toHaveStyle(`background: ${color.primary}`);
        expect(div).toHaveStyle(`border: 1px solid ${color.primary}`);
        expect(div).toHaveStyle(`border-radius: 40px`);
    });

    it("disabled button", () => {
        let wrapper = render(<Button disabled={true} round={true} appearance={"primary"}>disabled</Button>);
        expect(wrapper).toMatchSnapshot();
        let div = wrapper.queryByTestId("button-test");

        expect(div).toHaveTextContent(`disabled`);
        expect(div).toHaveStyle(`cursor: not-allowed`);
        expect(div).toHaveStyle(`color: rgba(0,0,0,.25)`);
        expect(div).toHaveStyle(`background: #f5f5f5`);
        expect(div).toHaveStyle(`border: 1px solid #d9d9d9`);
        expect(div).toHaveStyle(`border-radius: 40px`);
    });
});
