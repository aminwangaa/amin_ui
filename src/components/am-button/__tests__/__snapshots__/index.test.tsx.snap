// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component appearance button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm cNiMuR"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          primary
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm cNiMuR"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        primary
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component appearance button 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm kMTtRt"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          default
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm kMTtRt"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        default
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component appearance button 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm jxfAYt"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          success
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm jxfAYt"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        success
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component appearance button 4`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm prxeB"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          warning
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm prxeB"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        warning
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component appearance button 5`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm GJHMt"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          danger
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm GJHMt"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        danger
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component disabled button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm bQGFbN"
        data-testid="button-test"
        disabled=""
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          disabled
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm bQGFbN"
      data-testid="button-test"
      disabled=""
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        disabled
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component round button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm ctHGzB"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          primary round
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm ctHGzB"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        primary round
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component size button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm hLxDvr"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          small
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm hLxDvr"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        small
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component size button 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm cNiMuR"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          medium
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm cNiMuR"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        medium
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Button component size button 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-AxjAm hqcgnt"
        data-testid="button-test"
        href=""
        target=""
      >
        <span
          class="sc-AxirZ kvamqw"
        >
          large
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-AxjAm hqcgnt"
      data-testid="button-test"
      href=""
      target=""
    >
      <span
        class="sc-AxirZ kvamqw"
      >
        large
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
